File: web/src/plugin/datapermission/api/dataPermission.js
import service from '@/utils/request'

// 获取数据权限配置
export const getDataPermissionConfig = (data) => {
  return service({
    url: '/datapermission/getConfig?tableName=' + data.tableName + '&authorityId=' + data.authorityID,
    method: 'get',
  })
}

// 保存数据权限配置
export const saveDataPermissionConfig = (data) => {
  return service({
    url: '/datapermission/saveConfig',
    method: 'post',
    data
  })
}

// 获取数据库表列表
export const getDatabaseTables = () => {
  return service({
    url: '/datapermission/tableList',
    method: 'get'
  })
}

// 获取表字段信息
export const getTableColumns = (tableName) => {
  return service({
    url: '/datapermission/fieldList',
    method: 'get',
    params: { tableName }
  })
}

// 获取受控表列表
export const getControlledTables = () => {
  return service({
    url: '/datapermission/controlledTableList',
    method: 'get'
  })
}

// 创建受控表
export const createControlledTable = (data) => {
  return service({
    url: '/datapermission/controlledTable',
    method: 'post',
    data
  })
}

// 删除受控表
export const deleteControlledTable = (data) => {
  return service({
    url: '/datapermission/controlledTable',
    method: 'delete',
    data
  })
}

File: server/plugin/datapermission/router/data_permission.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/datapermission/api"
	"github.com/gin-gonic/gin"
)

type DataPermissionRouter struct{}

// InitDataPermissionRouter 初始化数据权限路由信息
func (s *DataPermissionRouter) InitDataPermissionRouter(Router *gin.RouterGroup) {
	dataPermissionRouter := Router.Use(middleware.OperationRecord())
	dataPermissionRouterWithoutRecord := Router
	dataPermissionApi := api.ApiGroupApp.DataPermissionApi
	{
		// 受控表管理
		dataPermissionRouter.POST("controlledTable", dataPermissionApi.CreateControlledTable)   // 创建受控表
		dataPermissionRouter.PUT("controlledTable", dataPermissionApi.UpdateControlledTable)    // 更新受控表
		dataPermissionRouter.DELETE("controlledTable", dataPermissionApi.DeleteControlledTable) // 删除受控表
	}
	{
		// 查询接口（不记录操作日志）
		dataPermissionRouterWithoutRecord.GET("controlledTableList", dataPermissionApi.GetControlledTableList) // 获取受控表列表
		dataPermissionRouterWithoutRecord.GET("getConfig", dataPermissionApi.GetDataPermissionConfig)          // 获取数据权限配置
		dataPermissionRouterWithoutRecord.GET("tableList", dataPermissionApi.GetTableList)                     // 获取数据库表列表
		dataPermissionRouterWithoutRecord.GET("fieldList", dataPermissionApi.GetFieldList)                     // 获取表字段列表
		dataPermissionRouterWithoutRecord.GET("statistics", dataPermissionApi.GetStatistics)                   // 获取统计信息
	}
	{
		// 配置管理
		dataPermissionRouter.POST("saveConfig", dataPermissionApi.SaveDataPermissionConfig) // 保存数据权限配置
	}
}
